gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -57,35 +59,67 @@ public ProcedureRequest mapToProcedureRequest(
             RCMRMT030101UKPlanStatement planStatement,
             Patient patient,
             List<Encounter> encounters,
-            String practiseCode
-    ) {
+            String practiseCode) {
 
         var id = planStatement.getId().getRoot();
-        var procedureRequest = new ProcedureRequest();
-        procedureRequest
-            .setStatus(getStatus(planStatement.getText()))
-            .setIntent(ProcedureRequestIntent.PLAN)
-            .setAuthoredOnElement(getAuthoredOn(planStatement.getAvailabilityTime(), ehrComposition))
-            .setOccurrence(getOccurrenceDate(planStatement.getEffectiveTime()))
-            .setSubject(new Reference(patient))
-            .setMeta(generateMeta(META_PROFILE))
-            .setId(id);
-        procedureRequest.getIdentifier().add(buildIdentifier(id, practiseCode));
-        procedureRequest.getNote().add(getNote(planStatement.getText()));
-        procedureRequest.setCode(codeableConceptMapper.mapToCodeableConcept(planStatement.getCode()));
-        procedureRequest.getRequester().setAgent(ParticipantReferenceUtil.getParticipantReference(planStatement.getParticipant(),
-            ehrComposition));
+        Meta meta = createMeta(ehrComposition, planStatement);
```

Comment:
I'm interested in your thinking of creating another method called `createMeta()` - I would say that this can just make a direct call to:

```java
final Meta meta = confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
            META_PROFILE,
            ehrComposition.getConfidentialityCode(),
            planStatement.getConfidentialityCode()
);
```

The reason I think this is because if there is only one location that `createMeta()` is called, following the YAGNI principle it should be excluded, and I don't believe it violates DRY  😄

Definitely not fussed about this, just more curious than anything!

Reply:
yeah, maybe you are right, it doesn't give much difference rather than being a wrapper of a call to createMetaAndAddSecurityIfConfidentialityCodesPresent method. I will inline it back into mapToProcedureRequest method